### PR TITLE
Refactor duplicate code in directives

### PR DESCRIPTION
**Pull request by JasperCraeghs**
Used as source by #106

### CODE REVIEW

File: mlx/directives/item_directive.py [hunk 1]
```diff
@@ -51,7 +53,7 @@ def perform_replacement(self, app, collection):
                         report_warning(env,
                                        'Traceability: relation {rel} cannot be translated to string'
                                        .format(rel=rel),
-                                       self.docname, self.line)
+                                       env.docname, self.line)
```
Comment (Letme) [inline mlx/directives/item_directive.py]: are we sure about this? Sounds like now warning might be on some other docname, or we simply do not init docname in class anymore?

File: mlx/traceable_base_directive.py [hunk 2]
```diff
@@ -0,0 +1,73 @@
+""" Module for the base class for all Coverity directives. """
+from abc import ABC, abstractmethod
+from docutils.parsers.rst import Directive
+
+from mlx.traceability_exception import report_warning
+from mlx.traceable_item import TraceableItem
+
+
+class BaseDirective(Directive, ABC):
+    """ Base class for all Coverity directives. """
```
Comment (Letme) [inline mlx/traceable_base_directive.py]: Traceability...

File: mlx/directives/item_directive.py (hunk 1 above)
Comment (JasperCraeghs) [reply mlx/directives/item_directive.py]: I am sure. I tried to init docname in the class as part of the refactoring, but it turned out to be of no use, so I reverted all but this instance. I should've reverted this on the other branch already.

File: mlx/traceable_base_directive.py (hunk 2 above)
Comment (JasperCraeghs) [reply mlx/traceable_base_directive.py]: 0b7d51d0175facbd499c6ece801ce6ddb561ab3b

**Pull request merged**